### PR TITLE
Split images into a base one and a default one

### PR DESCRIPTION
**Pull request by qgerome**

### CODE REVIEW

File: images/base/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM jupyter/datascience-notebook:2023-01-24
+FROM jupyter/datascience-notebook:2023-10-20
```
Comment (qgerome) [inline images/base/Dockerfile:1]: NOPE. It needs jupyterlab4

File: images/base/Dockerfile [hunk 2]
```diff
@@ -24,91 +24,26 @@ WORKDIR $HOME
...
-
 # /home/hexa symlink (preferred path for data pipelines)
 RUN ln -s /home/jovyan /home/hexa
 RUN chown jovyan /home/hexa
 
 USER ${NB_UID}
 
-# R packages (mamba + conda-forge)
-RUN mamba install --yes -c conda-forge \
-    'r-arrow=11.0.0' \
```
Comment (pvanliefland) [inline images/base/Dockerfile:38]: (keep)

File: images/base/Dockerfile [hunk 3]
```diff
@@ -24,91 +24,26 @@ WORKDIR $HOME
...
-    'r-aws.s3=0.*' \
-    'r-geojsonio=0.*' \
-    'r-getpass=0.*' \
```
Comment (pvanliefland) [inline images/base/Dockerfile:41]: (keep)

File: images/base/Dockerfile [hunk 4]
```diff
@@ -24,91 +24,26 @@ WORKDIR $HOME
...
-    'r-plotly=4.*' \
-    'r-raster=3.*' \
-    'r-readxl=1.*' \
-    'r-rcolorbrewer=1.*' \
-    'r-rgdal=1.*' \
-    'r-rgeos=0.*' \
-    'r-rgooglemaps=1.*' \
-    'r-rjava=1.*' \
-    'r-rjson=0.*' \
-    'r-rpostgres=1.*' \
```
Comment (pvanliefland) [inline images/base/Dockerfile:55]: (keep)

File: images/base/Dockerfile [hunk 5]
```diff
@@ -24,91 +24,26 @@ WORKDIR $HOME
...
-    'r-sf=1.*' \
-    'r-styler=1.*' \
```
Comment (pvanliefland) [inline images/base/Dockerfile:57]: (keep)

File: images/base/Dockerfile [hunk 6]
```diff
@@ -24,91 +24,26 @@ WORKDIR $HOME
...
-    'geopandas=0.*' \
-    'hybridcontents=0.*' \
-    'ipywidgets=7.*' \
     'isort=5.*' \
-    'lxml=4.*' \
-    'mapclassify=2.*' \
-    'nbresuse=0.*' \
-    'netCDF4=1.*' \
     'papermill==2.*' \
     'plotly=5.*' \
```
Comment (pvanliefland) [inline images/base/Dockerfile:40]: (remove)

File: images/base/Dockerfile [hunk 7]
```diff
@@ -24,91 +24,26 @@ WORKDIR $HOME
...
     'psycopg2=2.*' \
-    'pyarrow=11.0.0' \
```
Comment (pvanliefland) [inline images/base/Dockerfile:91]: (keep)

File: images/base/Dockerfile [hunk 8]
```diff
@@ -24,91 +24,26 @@ WORKDIR $HOME
...
-    'rapidfuzz=2.*' \
-    'rasterstats=0.*' \
-    'rpy2=3.*' \
-    's3contents=0.*' \
-    's3fs=2022.*' \
-    'slack_sdk=3.*' \
-    'xarray=2022.*' \
-    'zarr=2.*' \
-    'polars=0.*' \
```
Comment (qgerome) [inline images/base/Dockerfile:100]: put It back in the base image

File: images/base/Dockerfile [hunk 9]
```diff
@@ -24,91 +24,26 @@ WORKDIR $HOME
...
-RUN apt-get install -y libgdal-dev libudunits2-dev
-
 # /home/hexa symlink (preferred path for data pipelines)
 RUN ln -s /home/jovyan /home/hexa
 RUN chown jovyan /home/hexa
 
 USER ${NB_UID}
 
-# R packages (mamba + conda-forge)
-RUN mamba install --yes -c conda-forge \
```
Comment (qgerome) [inline images/base/Dockerfile:37]: The R deps must be in the default imageas well

**Review by pvanliefland — CHANGES_REQUESTED** [review]
I would:

- keep all R libraries (or almost)
- keep all existing libraries except
  - cowsay
  - s3contents
  - hybridcontents
  - fakeuseragent
  - rjava
- ask the DS team about
  - plotly
  - dask
- consider another name for `default`

**Comment by pvanliefland** [comment]
Other stuff:

- `dash`: should ask the DS team if they want to keep it
- `jupyter-server-proxy`: let's try to remove it

**Pull request merged**